test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -2080,6 +2077,115 @@ def test_cgroup_use_hierarchy(self):
+    if 'PBS_CONF_FILE' in os.environ:
+        pbs_conf_file = os.environ['PBS_CONF_FILE']
+    else:
+        pbs_conf_file = os.path.join(os.sep, 'etc', 'pbs.conf')
+    regex = re.compile(r'\\s*([^\\s]+)\\s*=\\s*([^\\s]+)\\s*')
+    try:
+        with open(pbs_conf_file, 'r') as desc:
+            for line in desc:
+                match = regex.match(line)
+                if match:
+                    if not pbs_home and match.group(1) == 'PBS_HOME':
+                        pbs_home = match.group(2)
+                    if not pbs_mom_home and (match.group(1) ==
+                                             'PBS_MOM_HOME'):
+                        pbs_mom_home = match.group(2)
+    except Exception:
+        pass
+if not pbs_home:
+    pbs.logmsg(pbs.EVENT_DEBUG, 'Failed to locate PBS_HOME')
+    event.reject()
+if not pbs_mom_home:
+    pbs_mom_home = pbs_home
+jobsfile = os.path.join(pbs_mom_home, 'mom_priv', 'hooks',
+                        'hook_data', 'cgroup_jobs')
+try:
+    with open(jobsfile, 'r+') as desc:
+        joblist = desc.readline().split()
+        jobset = set(joblist)
+        if jid_to_prepend not in jobset:
+            jobset.add(jid_to_prepend)
+            desc.seek(0)
+            desc.write(' '.join(jobset))
+            desc.truncate()
+except Exception as exc:
+    pbs.logmsg(pbs.EVENT_DEBUG, 'Failed to modify ' + jobsfile)
+    pbs.logmsg(pbs.EVENT_DEBUG, str(traceback.format_exc().strip().splitlines()))
+    event.reject()
+event.accept()
+""" % jid1
+        events = '"execjob_begin,exechost_periodic"'
+        hookconf = {'enabled': 'True', 'freq': 2, 'alarm': 30, 'event': events}
+        self.server.create_import_hook(hookname, hookconf, hookbody,
+                                       overwrite=True)
+        # Submit a second job and verify that the following message
+        # does NOT appear in the mom log:
+        # _exechost_periodic_handler: Failed to update jid1
+        presubmit = int(time.time())
+        a = {'Resource_List.select': '1:ncpus=1:mem=100mb:host=%s' %
+             self.hostA}
+        j = Job(TEST_USER, attrs=a)
+        j.create_script(self.sleep15_job)
+        jid2 = self.server.submit(j)
+        a = {'job_state': 'R'}
+        self.server.expect(JOB, a, jid2)
+        self.server.status(JOB, ATTR_o, jid2)
+        o = j.attributes[ATTR_o]
+        self.tempfile.append(o)
+        self.server.log_match(jid2 + ';Exit_status=0')
+        self.server.manager(MGR_CMD_DELETE, HOOK, None, hookname)
+        command = ['truncate', '-s0',
+                   os.path.join(self.momA.pbs_conf['PBS_HOME'], 'mom_priv',
+                                'hooks', 'hook_data', 'cgroup_jobs')]
+        self.du.run_cmd(cmd=command, hosts=self.hostA, sudo=True)
+        logmsg = '_exechost_periodic_handler: Failed to update %s' % jid1
+        self.momA.log_match(msg=logmsg, starttime=presubmit,
+                            max_attempts=1, existence=False)
```

Comment:
The test expects the "JobID;Exit_status=0" string in the server log before we check for any entries in the MoM log. Since we know the job has already completed, there is no race condition. Glad you asked.